### PR TITLE
Properly skip cleanup if only enlarging storage array.

### PR DESCRIPTION
**Pull request by chriseth**
Previously, the cleanup loop was reached and skipped right away, so this fix only makes it a little cheaper.

### CODE REVIEW

**Review by pirapira — APPROVED** [review]
Looks good to me.

**Comment by chriseth** [comment]
Waiting for a review by @axic (or a statement that it is fine to merge without a review from him).

**Pull request merged**
